utility/split-log-into-stages.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+set -u
+IFS=$'\n'
+
+cd "$(dirname "$1")"
+
+full_log_gz=$(basename "$1")
+gunzip --keep "$full_log_gz"
+full_log=$(basename "$1" .gz)
+
+function get_stage_regexp() {
+    echo '^.{29}(?:\[Pipeline\] )?\[(?!Pipeline)('"$1"')\].*$'
```

Comment:
> There could be one problem. Since regexps match eagerly, it could match some pipe inside a stage log.

Yes :( That's why I didn't do it. Perhaps matching `.{26} \| ` would be more readable? Or perhaps something like `(\s+|????-??-?? ??:??:??\.??????) \| `?

> I am a bit confused about (?!Pipeline). Could you explain what does it achieve?

It's called "negative lookahead" and it's purpose is so `[Pipeline]` tag is not counted as a stage. I've added a comment in following commit, what do you think?